src/simple.rs

```diff
@@ -46,7 +46,59 @@ impl SimpleClient {
         .await
     }
 
+    /// Execute redis command and stream response
+    pub fn stream<U>(
+        self,
+        cmd: U,
+    ) -> Result<impl Stream<Item = Result<U::Output, CommandError>>, CommandError>
+    where
+        U: Command,
+    {
+        self.io.encode(cmd.to_request(), &Codec)?;
+
+        let rs: RedisStream<U> = RedisStream {
+            io: self.io,
+            _cmd: std::marker::PhantomData,
+        };
+
+        Ok(rs)
+    }
+
     pub(crate) fn into_inner(self) -> IoBoxed {
         self.io
     }
 }
+
+pub struct RedisStream<U: Command> {
+    io: IoBoxed,
+    _cmd: std::marker::PhantomData<U>,
+}
+
+impl<U: Command> Stream for RedisStream<U> {
```

Comment:
I'd prefer to have `.recv()` and `.poll_recv()` in addition to Stream impl. those methods can use `&self`

Reply:
done